gtsfm/utils/ellipsoid.py

```diff
@@ -0,0 +1,179 @@
+        matrix.
+    """
+
+    # Iterate through each track to gather a list of 3D points forming the point cloud.
+    point_cloud_list = []
+    num_pts = gtsfm_data.number_tracks()
+
+    for j in range(num_pts):
+        track = gtsfm_data.get_track(j)  # TEMP: might need an extra import for track
+        x, y, z = track.point3()
+        point_cloud_list.append([x, y, z])
+
+    # Transform the point cloud to be aligned with x,y,z axes using SVD.
+    point_cloud = np.array([np.array(p) for p in point_cloud_list])  # point_cloud has shape Nx3
+    points_centered, means = center_point_cloud(point_cloud)
+    points_filtered = remove_outlier_points(points_centered)
+    wuprightRw = get_rotation_matrix(points_filtered)
+    aligned_points = apply_ellipsoid_rotation(wuprightRw, points_centered)
+
+    return aligned_points, means, wuprightRw
+
+
+def center_point_cloud(point_cloud: np.ndarray) -> Tuple[np.ndarray, np.ndarray]:
+    """Centers a point cloud using mean values of x, y, and z.
+
+    Args:
+        point_cloud: N 3D points to center, of shape N x 3.
+
+    Returns:
+        The centered point cloud of shape N x 3. Also an array of shape (3.) representing the mean x,y,z 
+        coordinates of point cloud.
+
+    Raises:
+        TypeError: if points are not of shape N x 3.
+    """
+    if point_cloud.shape[1] != 3:
+        raise TypeError("Points list should be 3D")
+
+    means = np.mean(point_cloud, axis=0)
+    points_centered = point_cloud - means
+    return points_centered, means
+
+
+def remove_outlier_points(point_cloud: np.ndarray) -> np.ndarray:
+    """Filters out the top 5% of points with greatest distance from origin.
+
+    Args:
+        point_cloud: point cloud of shape N x 3.
+
+    Returns:
+        The filtered point cloud of shape M x 3 (M = 0.95*N).
+
+    Raises:
+        TypeError: if centered point cloud is not of shape N x 3.
+    """
+    if point_cloud.shape[1] != 3:
+        raise TypeError("Point Cloud should be 3 dimensional")
+
+    # find mean and standard deviation of magnitudes of points to determine the threshold magnitude to filter by
+    mags = np.linalg.norm(point_cloud, axis=1)
+    mean_mag = np.mean(mags)
+    std_mag = np.std(mags)
+
+    # z = 1.65 represents the 95th percentile. Assuming a normal distribution of point magnitudes.
+    cutoff_mag = 1.65 * (std_mag) + mean_mag
+
+    indices_to_keep = np.where(np.linalg.norm(point_cloud, axis=1) < cutoff_mag)
+    points_filtered = point_cloud[indices_to_keep]
+    return points_filtered
+
+
+def get_rotation_matrix(point_cloud: np.ndarray) -> np.ndarray:
+    """Applies SVD on a point cloud. The resulting V contains the rotation matrix required to align the 3 major axes of
+    the point cloud's ellipsoid with the x, y, z coordinate axes.
+
+    Args:
+        point_cloud: point cloud of shape N x 3.
+
+    Returns:
+        The rotation matrix, shape 3 x 3, required to align points with the x, y, and z axes.
```

Comment:
maybe we can define what w and wupright are also.

Maybe `w` and `waligned` are better, since we have no guarantee its upright -- only guaranteed to be aligned to orthonormal axes.

i.e. World and axis-aligned world frames